gpii/node_modules/settingsHandlers/src/WebSocketsComponent.js

```diff
@@ -0,0 +1,262 @@
+        modelListeners: {
+            "clients.*": {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.clientsChanged",
+                args: "{change}"
+            },
+            "settings.*": {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.settingsChanged",
+                args: "{change}"
+            }
+        },
+        invokers: {
+            addClient: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.addClient",
+                args: ["{that}", "{arguments}.0", "{arguments}.1"]
+            },
+            removeClient: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.removeClient",
+                args: ["{that}", "{arguments}.0"]
+            },
+            getSettingsForId: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.getSettingsForId",
+                args: ["{that}", "{arguments}.0"]
+            },
+            setSettingsForId: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.setSettingsForId",
+                args: ["{that}", "{arguments}.0", "{arguments}.1"]
+            },
+            //getSettings: {
+            //    funcName: "gpii.settingsHandlers.WebSocketsComponent.getSettings",
+            //    args: ["{arguments}.0", "{arguments}.1", "{that}"]
+            //},
+            //setSettingsIndividually: {
+            //    funcName: "gpii.settingsHandlers.WebSocketsComponent.setSettingsIndividually",
+            //    args: ["{arguments}.0", "{arguments}.1", "{that}"]
+            //},
+            //setSettings: {
+            //    funcName: "gpii.settingsHandlers.WebSocketsComponent.setSettings",
+            //    args: ["{arguments}.0", "{arguments}.1", "{that}"]
+            //},
+            get: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.get",
+                args: ["{arguments}.0", "{that}"]
+            },
+            set: {
+                funcName: "gpii.settingsHandlers.WebSocketsComponent.set",
+                args: ["{arguments}.0", "{that}"]
+            }
+        }
+    });
+
+    ///////////////// Clients /////////////////////
+
+    gpii.settingsHandlers.WebSocketsComponent.clientsChanged = function (a) {
+        var util = require("util");
+        console.log("A change in clients has been registered: " + util.inspect(a));
+    }
+
+    gpii.settingsHandlers.WebSocketsComponent.addClient = function (that, solutionId, client) {
+       var solution = solutionId.replace(/\./g, "\\.");
+       var type = fluid.get(that.model.clients, [solutionId, client.id]) === undefined ? "ADD": null;
+
+       // How can I set these options to the applier in a declarative way?
+       //
+       that.applier.options.resolverGetConfig = fluid.model.escapedGetConfig;
+       that.applier.options.resolverSetConfig = fluid.model.escapedSetConfig;
+
+       that.applier.change("clients." + solution + "." + client.id, client, type);
```

Comment:
On 15/06/2015 18:16, Javier Hernández Antúnez wrote:
> In gpii/node_modules/settingsHandlers/src/WebSocketsComponent.js
> <https://github.com/GPII/universal/pull/303#discussion_r32443534>:
>
>> +
>> +    gpii.settingsHandlers.WebSocketsComponent.clientsChanged = function (a) {
>> +        var util = require("util");
>> +        console.log("A change in clients has been registered: " + util.inspect(a));
>> +    }
>> +
>> +    gpii.settingsHandlers.WebSocketsComponent.addClient = function (that, solutionId, client) {
>> +       var solution = solutionId.replace(/\./g, "\\.");
>> +       var type = fluid.get(that.model.clients, [solutionId, client.id]) === undefined ? "ADD": null;
>> +
>> +       // How can I set these options to the applier in a declarative way?
>> +       //
>> +       that.applier.options.resolverGetConfig = fluid.model.escapedGetConfig;
>> +       that.applier.options.resolverSetConfig = fluid.model.escapedSetConfig;
>> +
>> +       that.applier.change("clients." + solution + "." + client.id, client, type);
>
> @amb26 <https://github.com/amb26>, by changing it to modelRelayComponent now I'm getting a ``` FATAL ERROR:
> Uncaught exception: Maximum call stack size exceeded ``` when calling to _that.applier.change_. Not sure if
> it's me doing something wrong, is there any relevant difference between the use of modelRelayComponent and
> modelComponent that needs to be taken into account when using it? 2015-06-15 17:43 GMT+02:00 Antranig Basman
> <notifications@github.com>:
> … <#>
> In gpii/node_modules/settingsHandlers/src/WebSocketsComponent.js
> <https://github.com/GPII/universal/pull/303#discussion_r32434020>: > + > +
> gpii.settingsHandlers.WebSocketsComponent.clientsChanged = function (a) { > + var util = require("util"); >
> + console.log("A change in clients has been registered: " + util.inspect(a)); > + } > + > +
> gpii.settingsHandlers.WebSocketsComponent.addClient = function (that, solutionId, client) { > + var solution
> = solutionId.replace(/\./g, "\\."); > + var type = fluid.get(that.model.clients, [solutionId, client.id])
> === undefined ? "ADD": null; > + > + // How can I set these options to the applier in a declarative way? > +
> // > + that.applier.options.resolverGetConfig = fluid.model.escapedGetConfig; > +
> that.applier.options.resolverSetConfig = fluid.model.escapedSetConfig; > + > +
> that.applier.change("clients." + solution + "." + client.id, client, type); OK, sorry about this. This is a
> fault in the very ancient Infusion ChangeApplier implementation - if you make a change from
> "fluid.modelComponent" to "fluid.modelRelayComponent" on line 24 of WebSocketsComponent.js it will work - I
> tested this with the revision of infusion currently in use in the GPII — Reply to this email directly or
> view it on GitHub <https://github.com/GPII/universal/pull/303/files#r32434020>.
> -- Javier Hernández Antúnez Área de Operaciones Emergya Consultoría Tlfno: +34 954 51 75 77 Fax: +34 954 51
> 64 73 http://www.emergya.es
>
> —
> Reply to this email directly or view it on GitHub <https://github.com/GPII/universal/pull/303/files#r32443534>.
>

I can't find this comment in the github interface.

It works ok for me: https://github.com/amb26/universal/tree/GPII-989 - what are you doing when you receive 
the overflow?